### PR TITLE
Close prepared statement used to set the lock timeout

### LINKED ISSUE #16801: Log message about open
### Before reporting an issue

- [X] I have searched existing issues
- [X] I have reproduced the issue with the latest release

### Area

storage

### Describe the bug

The PR https://github.com/keycloak/keycloak/pull/16524 introduced a bug where a prepared statement wasn't closed when the lock timeout was configured and when running on Quarkus:

> [io.agroal.pool] (Quarkus Main Thread) Datasource '<default>': JDBC resources leaked: 0 ResultSet(s) and 1 Statement(s)

### Version

nightly

### Expected behavior

Statement should be closed

### Actual behavior

Statement isn't closed by the code, but by the connection pool.

### How to Reproduce?

Start Quarkus from the nightly build with JPA map and the timeout set (KC_SPI_MAP_STORAGE_JPA_LOCK_TIMEOUT=1000).

### Anything else?

_No response_

### PR DESCRIPTION
**Pull request by ahus1**
Closes #16801

### CODE REVIEW

File: model/map-jpa/src/main/java/org/keycloak/models/map/storage/jpa/JpaMapStorageProviderFactory.java [hunk 1]
```diff
@@ -288,8 +288,10 @@ protected EntityManager getEntityManager() {
             if (lockTimeout != null) {
                 em.unwrap(SessionImpl.class)
                         .doWork(connection -> {
-                            PreparedStatement preparedStatement = connection.prepareStatement("SET LOCAL lock_timeout = '" + lockTimeout + "';");
-                            preparedStatement.execute();
+                            try (PreparedStatement preparedStatement = connection.prepareStatement("SET LOCAL lock_timeout = ?")) {
+                                preparedStatement.setString(1, String.valueOf(lockTimeout));
```
Comment (mhajas) [inline model/map-jpa/src/main/java/org/keycloak/models/map/storage/jpa/JpaMapStorageProviderFactory.java:292]: I tried to use parameters here, however, it seemed that something (Postgres/Hibernate) support paremeters only for some SQL commands and the way I understood it `SET` is not between them. But would be great if you were able to make it work.

Comment (ahus1) [reply model/map-jpa/src/main/java/org/keycloak/models/map/storage/jpa/JpaMapStorageProviderFactory.java:292]: > I tried to use parameters here, however, it seemed that something (Postgres/Hibernate) support parameters only for some SQL commands and the way I understood it SET is not between them. But would be great if you were able to make it work.

I've seen it work for CRDB, and now see it failing for PostgreSQL on GHA. Can only try it locally next week again. Will add comment that parameters won't work here. 

BTW: This is plain PostgreSQL JDBC, no Hibernate involved here as we're asking for a direct JDBC connection.

Comment (mhajas) [reply model/map-jpa/src/main/java/org/keycloak/models/map/storage/jpa/JpaMapStorageProviderFactory.java:292]: Ahh, sure, that makes sense. Thanks for the explanation. I should have added the comment myself, sorry. Good luck with the conference btw. Hope you have a pleasant journey!

Comment (ahus1) [reply model/map-jpa/src/main/java/org/keycloak/models/map/storage/jpa/JpaMapStorageProviderFactory.java:293]: Found a way with a prepared statement and a parameter. Build is now green with PostgreSQL, will test on Monday with CRDB, and then ask you for a review.

**Comment by ahus1** [comment]
@mhajas - tested it with CRDB, builds are green. Ready for review. Thanks!

File: model/map-jpa/src/main/java/org/keycloak/models/map/storage/jpa/JpaMapStorageProviderFactory.java [hunk 2]
```diff
@@ -18,11 +18,7 @@
 
 import static org.keycloak.models.map.storage.jpa.updater.MapJpaUpdaterProvider.Status.VALID;
 
-import java.sql.Connection;
-import java.sql.DatabaseMetaData;
-import java.sql.DriverManager;
-import java.sql.PreparedStatement;
-import java.sql.SQLException;
+import java.sql.*;
```
Comment (mhajas) [inline model/map-jpa/src/main/java/org/keycloak/models/map/storage/jpa/JpaMapStorageProviderFactory.java:21]: As far as I know we don't use asterisk imports in Keycloak codebase, can you please revert?

**Review by mhajas — CHANGES_REQUESTED** [review]
Thank you for the PR, I added one codestyle comment, could you please have a look?

**Review by mhajas — APPROVED** [review]
Thanks for the fix @ahus1! LGTM

**Pull request merged**